store/store_test.go

```diff
@@ -3261,25 +3592,57 @@ func (s *storeTestSuite) TestFind500(c *C) {
 	cfg := store.Config{
 		StoreBaseURL: mockServerURL,
 		DetailFields: []string{},
+		SearchFields: []string{},
 	}
 	sto := store.New(&cfg, nil)
 
 	_, err := sto.Find(s.ctx, &store.Search{Query: "hello"}, nil)
 	c.Check(err, ErrorMatches, `cannot search: got unexpected HTTP status code 500 via GET to "http://\S+[?&]q=hello.*"`)
 	c.Assert(n, Equals, 5)
+	if apiV1 {
+		c.Check(v1Fallback, Equals, true)
+	} else {
+		c.Check(v2Hit, Equals, true)
+	}
 }
 
-func (s *storeTestSuite) TestFind500once(c *C) {
+func (s *storeTestSuite) TestFindV1_500(c *C) {
+	apiV1 := true
+	s.testFind500(c, apiV1)
+}
+
+func (s *storeTestSuite) TestFindV2_500(c *C) {
+	s.testFind500(c, false)
+}
+
+func (s *storeTestSuite) testFind500Once(c *C, apiV1 bool) {
```

Comment:
do you need what's the difference between Find500 and Find500Once ? maybe given we are renaming things we can give these a better name?

Reply:
Yes, renamed and added comment explaining the difference.